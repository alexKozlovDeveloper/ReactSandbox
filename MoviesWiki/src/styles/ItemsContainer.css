.container {
    display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	align-items: stretch;
    align-content: stretch;    
    width: 100%;
}