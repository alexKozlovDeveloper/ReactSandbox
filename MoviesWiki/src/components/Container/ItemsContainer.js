import React, { Component } from "react";

import ItemBody from "./ItemBody";
import NoItemLabel from "./NoItemLabel";

import styles from "../../styles/ItemsContainer.css"

import { renderToString } from 'react-dom/server'

class ItemsContainer extends Component {

    render() {
        
        function getDivTable(items, itemsPerRow) {
            var tableContent = [];     

            for (var i = 0, rowIndex = 1; i < items.length; i++, rowIndex++) {        
                tableContent.push(<ItemBody key={i} title={items[i].title} genre={items[i].genre} image={items[i].image} releaseDate={items[i].releaseDate}/>);        
            }
        
            return tableContent;
        }

        var table = "";

        if(this.props.items.length !== 0) {
            table = getDivTable(this.props.items, parseInt(this.props.itemsPerRow));
        } else{            
            table = renderToString(<NoItemLabel />);
        }

        return (
            <div className={styles.container} >
                {table}
            </div>
        );
    }
}

export default ItemsContainer;