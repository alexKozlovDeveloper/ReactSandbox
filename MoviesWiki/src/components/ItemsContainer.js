import React, { Component } from "react";

import ItemBody from "./ItemBody";

import styles from "../styles/ItemsContainer.css"

class ItemsContainer extends Component {
    render() {
        return (
            <div className={styles.container}>
            <table>
                <tr>
                    <td>
                        <ItemBody title="Kill Bill 2" genre="Action" image="https://st.kp.yandex.net/images/film_big/2717.jpg" releaseDate="2003" />
                    </td>
                    <td>
                        <ItemBody title="Kill Bill 2" genre="Action" image="https://st.kp.yandex.net/images/film_big/2717.jpg" releaseDate="2003" />
                    </td>
                    <td>
                        <ItemBody title="Kill Bill 2" genre="Action" image="https://st.kp.yandex.net/images/film_big/2717.jpg" releaseDate="2003" /> 
                    </td>
                </tr>
            </table>
            </div>
        );
    }
}

export default ItemsContainer;