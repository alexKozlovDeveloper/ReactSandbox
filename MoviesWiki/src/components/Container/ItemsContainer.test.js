import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import rootReducer from '../../reducers'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import { BrowserRouter as Router } from 'react-router-dom'

import ItemsContainer from "./ItemsContainer";

configure({ adapter: new Adapter() });

describe('ItemsContainer component', () => {

    it('Rendering test - empty', () => {
        var moviesList = [];

        var resultsBodyConfig = {
            itemsPerRow: 4,
            resultSortConfig: {
                title: "Sort by",
                selectedIndex: "1",
                options: ["Release date", "Rating"]
            },
            resultsCountConfig: {
                title: "movies found"
            }
        };

        const store = createStore(rootReducer)

        const component = mount(<Router><Provider store={store}><ItemsContainer items={moviesList} config={resultsBodyConfig} /></Provider></Router>);
        expect(shallowToJson(component)).toMatchSnapshot();
    });

    it('Rendering test - with items', () => {
        var moviesList = [{
            title: "Interstellar",
            releaseDate: "2014",
            image: "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            genres: ["action"],
            rating: 7.2,
            duration: 145,
            description: "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
        }, {
            title: "Interstellar 2",
            releaseDate: "2015",
            image: "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            genres: ["action2"],
            rating: 7.2,
            duration: 145,
            description: "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
        }, {
            title: "Interstellar 3",
            releaseDate: "2016",
            image: "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            genres: ["action 3"],
            rating: 7.2,
            duration: 145,
            description: "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
        }];

        var resultsBodyConfig = {
            itemsPerRow: 4,
            resultSortConfig: {
                title: "Sort by",
                selectedIndex: "1",
                options: ["Release date", "Rating"]
            },
            resultsCountConfig: {
                title: "movies found"
            }
        };

        const store = createStore(rootReducer)

        const component = mount(<Router><Provider store={store}><ItemsContainer items={moviesList} config={resultsBodyConfig} /></Provider></Router>);
        expect(shallowToJson(component)).toMatchSnapshot();
    });

})

