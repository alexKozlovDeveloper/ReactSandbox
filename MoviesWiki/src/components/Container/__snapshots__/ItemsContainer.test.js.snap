// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsContainer component Rendering test - empty 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ItemsContainer
    config={
      Object {
        "itemsPerRow": 4,
        "resultSortConfig": Object {
          "options": Array [
            "Release date",
            "Rating",
          ],
          "selectedIndex": "1",
          "title": "Sort by",
        },
        "resultsCountConfig": Object {
          "title": "movies found",
        },
      }
    }
    items={Array []}
  >
    <div
      className="container"
    >
      <NoItemLabel>
        <div
          className="title"
        >
          No films found
        </div>
      </NoItemLabel>
    </div>
  </ItemsContainer>
</Provider>
`;

exports[`ItemsContainer component Rendering test - with items 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ItemsContainer
    config={
      Object {
        "itemsPerRow": 4,
        "resultSortConfig": Object {
          "options": Array [
            "Release date",
            "Rating",
          ],
          "selectedIndex": "1",
          "title": "Sort by",
        },
        "resultsCountConfig": Object {
          "title": "movies found",
        },
      }
    }
    items={
      Array [
        Object {
          "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
          "duration": 145,
          "genres": Array [
            "action",
          ],
          "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
          "rating": 7.2,
          "releaseDate": "2014",
          "title": "Interstellar",
        },
        Object {
          "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
          "duration": 145,
          "genres": Array [
            "action2",
          ],
          "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
          "rating": 7.2,
          "releaseDate": "2015",
          "title": "Interstellar 2",
        },
        Object {
          "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
          "duration": 145,
          "genres": Array [
            "action 3",
          ],
          "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
          "rating": 7.2,
          "releaseDate": "2016",
          "title": "Interstellar 3",
        },
      ]
    }
  >
    <div
      className="container"
    >
      <Connect(ItemBody)
        item={
          Object {
            "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
            "duration": 145,
            "genres": Array [
              "action",
            ],
            "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            "rating": 7.2,
            "releaseDate": "2014",
            "title": "Interstellar",
          }
        }
        key="0"
      >
        <ItemBody
          item={
            Object {
              "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
              "duration": 145,
              "genres": Array [
                "action",
              ],
              "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
              "rating": 7.2,
              "releaseDate": "2014",
              "title": "Interstellar",
            }
          }
          updateSelectedItemFunc={[Function]}
        >
          <div
            className="body"
            onClick={[Function]}
          >
            <Image>
              <div
                className="imagecontainer"
              >
                <img
                  className="image"
                />
              </div>
            </Image>
            <ReleaseDate>
              <div
                className="releasedate"
              />
            </ReleaseDate>
            <Title
              title="Interstellar"
            >
              <div
                className="title"
              >
                Interstellar
              </div>
            </Title>
            <Genre
              genre="action"
            >
              <div
                className="genre"
              >
                action
              </div>
            </Genre>
          </div>
        </ItemBody>
      </Connect(ItemBody)>
      <Connect(ItemBody)
        item={
          Object {
            "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
            "duration": 145,
            "genres": Array [
              "action2",
            ],
            "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            "rating": 7.2,
            "releaseDate": "2015",
            "title": "Interstellar 2",
          }
        }
        key="1"
      >
        <ItemBody
          item={
            Object {
              "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
              "duration": 145,
              "genres": Array [
                "action2",
              ],
              "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
              "rating": 7.2,
              "releaseDate": "2015",
              "title": "Interstellar 2",
            }
          }
          updateSelectedItemFunc={[Function]}
        >
          <div
            className="body"
            onClick={[Function]}
          >
            <Image>
              <div
                className="imagecontainer"
              >
                <img
                  className="image"
                />
              </div>
            </Image>
            <ReleaseDate>
              <div
                className="releasedate"
              />
            </ReleaseDate>
            <Title
              title="Interstellar 2"
            >
              <div
                className="title"
              >
                Interstellar 2
              </div>
            </Title>
            <Genre
              genre="action2"
            >
              <div
                className="genre"
              >
                action2
              </div>
            </Genre>
          </div>
        </ItemBody>
      </Connect(ItemBody)>
      <Connect(ItemBody)
        item={
          Object {
            "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
            "duration": 145,
            "genres": Array [
              "action 3",
            ],
            "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            "rating": 7.2,
            "releaseDate": "2016",
            "title": "Interstellar 3",
          }
        }
        key="2"
      >
        <ItemBody
          item={
            Object {
              "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
              "duration": 145,
              "genres": Array [
                "action 3",
              ],
              "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
              "rating": 7.2,
              "releaseDate": "2016",
              "title": "Interstellar 3",
            }
          }
          updateSelectedItemFunc={[Function]}
        >
          <div
            className="body"
            onClick={[Function]}
          >
            <Image>
              <div
                className="imagecontainer"
              >
                <img
                  className="image"
                />
              </div>
            </Image>
            <ReleaseDate>
              <div
                className="releasedate"
              />
            </ReleaseDate>
            <Title
              title="Interstellar 3"
            >
              <div
                className="title"
              >
                Interstellar 3
              </div>
            </Title>
            <Genre
              genre="action 3"
            >
              <div
                className="genre"
              >
                action 3
              </div>
            </Genre>
          </div>
        </ItemBody>
      </Connect(ItemBody)>
    </div>
  </ItemsContainer>
</Provider>
`;
